ext/dom/element.c

```diff
@@ -728,57 +850,37 @@ PHP_METHOD(DOMElement, getAttributeNS)
 
 	DOM_GET_OBJ(elemp, id, xmlNodePtr, intern);
 
-	strattr = xmlGetNsProp(elemp, (xmlChar *) name, (xmlChar *) uri);
-
-	if (strattr != NULL) {
-		RETVAL_STRING((char *)strattr);
-		xmlFree(strattr);
+	bool should_free_result;
+	const xmlChar *result = dom_get_attribute_ns(intern, elemp, uri, uri_len, name, &should_free_result);
+	if (result == NULL) {
+		if (php_dom_follow_spec_intern(intern)) {
+			RETURN_NULL();
+		}
+		RETURN_EMPTY_STRING();
 	} else {
-		if (xmlStrEqual((xmlChar *) uri, (xmlChar *)DOM_XMLNS_NAMESPACE)) {
-			nsptr = dom_get_nsdecl(elemp, (xmlChar *)name);
-			if (nsptr != NULL) {
-				RETVAL_STRING((char *) nsptr->href);
-			} else {
-				RETVAL_EMPTY_STRING();
-			}
-		} else {
-			RETVAL_EMPTY_STRING();
+		RETVAL_STRING((const char *) result);
+		if (should_free_result) {
+			xmlFree(BAD_CAST result);
 		}
 	}
-
 }
 /* }}} end dom_element_get_attribute_ns */
 
-/* {{{ URL: http://www.w3.org/TR/2003/WD-DOM-Level-3-Core-20030226/DOM3-Core.html#core-ID-ElSetAttrNS
-Since: DOM Level 2
-*/
-PHP_METHOD(DOMElement, setAttributeNS)
+static void dom_set_attribute_ns_legacy(dom_object *intern, xmlNodePtr elemp, char *uri, char *name, const char *value, size_t uri_len, size_t name_len)
```

Comment:
```suggestion
static void dom_set_attribute_ns_legacy(dom_object *intern, xmlNodePtr elemp, char *uri, size_t uri_len, char *name, size_t name_len, const char *value)
```
Maybe? Or is there a reason for them to be at the end?

Or use zend_strings ?

Reply:
I can reorder them. I could also use zend_string but this is old DOM code that I factored out so I didn't want to refactor the contents too as that would make the PR even more heavy.

Reply:
Makes sense, can always be later done